ecs-agent/api/container/restart/restart_tracker.go

```diff
@@ -0,0 +1,71 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package restart
+
+import (
+	"fmt"
+	"time"
+
+	api "github.com/aws/amazon-ecs-agent/ecs-agent/acs/model/ecsacs"
+	apicontainerstatus "github.com/aws/amazon-ecs-agent/ecs-agent/api/container/status"
+
+	"github.com/aws/aws-sdk-go/aws"
+)
+
+type RestartTracker struct {
+	RestartCount  int `json:"restartCount,omitempty"`
+	restartPolicy *api.ContainerRestartPolicy
+}
+
+func NewRestartTracker(restartPolicy *api.ContainerRestartPolicy) *RestartTracker {
+	return &RestartTracker{
+		restartPolicy: restartPolicy,
+	}
+}
+
+func (rt *RestartTracker) GetRestartCount() int {
```

Comment:
should we put a mutex with a read lock here and and a write lock in RecordRestart()?

Reply:
can do, have added this